tsconfig.json

```diff
@@ -45,7 +45,8 @@
 
     /* Experimental Options */
     "experimentalDecorators": true,           /* Enables experimental support for ES7 decorators. */
-    "emitDecoratorMetadata": true             /* Enables experimental support for emitting type metadata for decorators. */
+    "emitDecoratorMetadata": true,             /* Enables experimental support for emitting type metadata for decorators. */
+    "outDir": "dist"
```

Comment:
@mslipper What is the advantage of `dist` folder?
To mix ts/js files, to retain the filenames, was intentional. We compile TS files to JS. This is a burden already. I would like not to make life even more complicated by asking a user to think about special `dist` folder.

Reply:
It makes it easier to use Machinomy in other TypeScript projects. If declaration files and source files are in the same directory, the TypeScript compiler resolves imports to the source file and will attempt to compile it. This means that other modules that depend on Machinomy will have to use a `tsconfig` file and TypeScript version that's compatible with Machinomy's in order to compile their application.

Putting the declaration files and JS sources files in dist should be transparent to everyone that imports Machinomy directly, since the package.json file points `main` to `./dist/index.js` and `typings` to `./dist/index.d.ts`. For users who want to import classes from the Machinomy lib (like I do for the payment hub), the only difference is the addition of `dist` to the import path.

Reply:
Okay, makes sense.